### DISCUSSION TITLE
How to use Citation and Attribution in chatflow？

### DISCUSSION THREAD

**Discussion by AnomeZ** [langgenius/dify#7635] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

When I was using the chatflow and test the knowledgebase, I found the chatflow didn't show the **Citations and Attributions** , while I actually did open this feature.

![image](https://github.com/user-attachments/assets/678b1d60-7a6d-4bea-9b44-b707005c814e)
![image](https://github.com/user-attachments/assets/38c819e8-65c2-45fd-901f-8a8c5c543f84)

As you can see, it should theoretically show the file below the conversation bubble. _(I don't know what to call it, forgive me for my broken English)_

I checked the **Retrieval Test**, It shows that the conversation did call the knowledge base, but somehow didn't return attribution. 
![image](https://github.com/user-attachments/assets/a45d6cb3-9971-46d3-b909-99d9c794eb40)

And this is the chatflow:
![image](https://github.com/user-attachments/assets/06be9560-7a1d-4443-b8a2-cee2ab30cb9d)

### 2. Additional context or comments

Deployment mode: on-premise
Use llm: llama3.1
Functions: Chatflow, Citation
Knowledge Base: Hybrid Search using the reranker model
Error message: None

**Comment by lvbibir** [comment]
I have also encountered this problem, have you managed to solve it?

**Comment by alliecatowo** [comment]
same issue, I feel like it can't display the citations since its just a string output

**Comment by mozhou-tech** [comment]
same issue.

**Comment by Reverse-Flash-Kamen** [comment]
You need to use the Knowledge Retrieval node's output {{#context#}} in the LLM Node.

![微信图片_20250609110949](https://github.com/user-attachments/assets/d14e3eed-7c02-4170-8975-f9b56091a00a)

**Reply by applepieiris** [reply]
not this problem, the author want to see the citations
